cmd/pilot-agent/main.go

```diff
@@ -97,14 +84,62 @@ var (
 				}
 			}
 
-			glog.V(2).Infof("proxy role: %#v", role)
+			glog.V(2).Infof("Proxy role: %#v", role)
 
-			envoyProxy := envoy.MakeProxy(mesh, servicecluster, role.ServiceNode(), configpath, binarypath)
-			agent := proxy.NewAgent(envoyProxy, proxy.DefaultRetry)
-			watcher, err := envoy.NewWatcher(mesh, agent, role)
-			if err != nil {
+			proxyConfig := proxy.DefaultProxyConfig()
+			if config != "" && configFile != "" {
+				return errors.New("cannot specify both configuration content and file at the same time")
```

Comment:
I still don't think passing a yaml file as command line parameter is reasonable - and I'm not aware of any other app doing this. 

--config "foo: bar\n  blah: | foo\n" etc is not readable, doesn't look good when you do "ps", and it is completely 
unneeded and unexpected for a user. We use yaml config files everywhere else.

Please remove this.

Reply:
Can we treat this as an implementation detail for Admission Controller? I agree it looks awful, but that's the easier way to pass config from AC to pods. Otherwise, we have to allocate space to write files and then distribute those files to each pod.

Reply:
I added configFile for the normal case

Reply:
s/YAML/JSON and the readability issue with newlines goes away? Its functionally equivalent albeit still a bit verbose.

Reply:
BTW - there is also viper ( same repo as cobra ) - that adds support for watching/reloading configs, env, and a bunch of formats - and works well with cobra flags.

Reply:
Files are difficult in the container world. Pod template is the best place to put these configs, and embedded YAML looks pretty nice actually.

Reply:
@costinm what is your constructive solution that works for k8s AC? let's not block PRs with generic criticisms.

Reply:
Just change to flags? Keep the config file option around.

Reply:
Ok, that can be done later, right. I'm trying to get to the state that mesh config is dynamic because that's a functional improvement. The switch from inline config from flags is a mechanical conversion process. I don't believe agent is part of our public API, it's an implementation detail of how Pilot operates.